### PR TITLE
v2 Delete cleanup

### LINKED ISSUE #5559: remove `odo v2delete`
more info in #5521 

## Acceptance criteria 
- [x] test cases for the removed command should be removed, tests cases that use removed commands should be updated
- [x] commands that are going to be removed need to be removed together with their implementation and test cases
- [ ] all help messages and error messages need to be reviewed and updated to make sure that it doesn't contain a reference to commands that no longer exist
- [x] remove `Delete` function in https://github.com/redhat-developer/odo/blob/a8218725e4e556b76e11ca2194a500bc0f4dee32/pkg/component/component.go#L547

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/kind cleanup

**What does this PR do / why we need it:**
This PR removes references to odo v2delete and cleans up unused functions and tests.
**Which issue(s) this PR fixes:**

Fixes #5559 

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
Relevant integration tests from v2delete have been added to v3 delete.

### CODE REVIEW

File: pkg/odo/genericclioptions/runnable.go [hunk 1]
```diff
@@ -160,27 +160,7 @@ func checkConflictingFlags(cmd *cobra.Command, args []string) error {
...
-				if project != "" && componentName == "" && app == "" {
-					return fmt.Errorf("cannot provide --project without --app and component name")
-				}
-				if all && ((componentName != "" && app != "" && project != "") || (componentName != "")) {
-					return fmt.Errorf("cannot provide --all when component name, --app and --project are provided")
-				}
-			}
-		}
-	}
+	// TODO: add validations for the delete command
```
Comment (dharmit) [inline pkg/odo/genericclioptions/runnable.go:163]: Will this be added as a part of this PR or a separate issue/PR?

File: tests/integration/devfile/cmd_delete_test.go [hunk 2]
```diff
@@ -229,4 +229,28 @@ ComponentSettings:
 			})
 		})
 	})
+	FWhen("deleting a component containing preStop event that is deployed with DEV", func() {
```
Comment (dharmit) [inline tests/integration/devfile/cmd_delete_test.go:232]:
```suggestion
	When("deleting a component containing preStop event that is deployed with DEV", func() {
```

**Review by dharmit — CHANGES_REQUESTED** [review]

File: tests/integration/devfile/cmd_devfile_create_test.go [hunk 3]
```diff
@@ -255,12 +255,16 @@ var _ = Describe("odo devfile create command tests", func() {
 	})
 
 	When("devfile exists in the working directory", func() {
+		var workingDir string
 		BeforeEach(func() {
+			workingDir = helper.Getwd()
```
Comment (feloy) [inline tests/integration/devfile/cmd_devfile_create_test.go:260]: Why are you doing this? As I understand, workingDir == commonVar.Context

**Review by feloy — CHANGES_REQUESTED** [review]

File: pkg/odo/genericclioptions/runnable.go (hunk 1 above)
Comment (valaparthvi) [reply pkg/odo/genericclioptions/runnable.go:163]: Separate PR/issue.

Comment (valaparthvi) [reply pkg/odo/genericclioptions/runnable.go]: Or perhaps this is not required. I have removed the TODO for now. We can add more validations as and when they're encountered.

File: tests/integration/devfile/cmd_devfile_deploy_test.go [hunk 4]
```diff
@@ -41,13 +41,13 @@ var _ = Describe("odo devfile deploy command tests", func() {
...
 		deploymentName := "my-component"
-
 		BeforeEach(func() {
 			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
 		})
 		AfterEach(func() {
-			helper.Cmd("odo", "v2delete", "-af").ShouldPass()
+			// assuming we are inside the context directory since `odo deploy` cannot be run outside a context directory
+			helper.Cmd("odo", "delete", "component", "-f").ShouldPass()
```
Comment (feloy) [inline tests/integration/devfile/cmd_devfile_deploy_test.go:50]: These delete are part of the cleanup, which is handled by deleting the namespace, I think it is not necessary and would be cleaner to not run the `odo delete` here (and in other places in the file)

File: tests/integration/devfile/cmd_devfile_deploy_test.go [hunk 5]
```diff
@@ -98,13 +98,13 @@ var _ = Describe("odo devfile deploy command tests", func() {
...
 	When("using a devfile.yaml containing two deploy commands", func() {
-
 		BeforeEach(func() {
 			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-two-deploy-commands.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
 		})
 		AfterEach(func() {
-			helper.Cmd("odo", "v2delete", "-a").ShouldPass()
+			// assuming we are inside the context directory since `odo deploy` cannot be run outside a context directory
+			helper.Cmd("odo", "delete", "component", "-f").ShouldPass()
```
Comment (dharmit) [inline tests/integration/devfile/cmd_devfile_deploy_test.go:107]: IMO, `AfterEach` should help aid cleanup part. If something in it has to have `ShouldPass`, it should either be in the test or we should not have `ShouldPass` call to it. Basically, `AfterEach` should not lead to a test failure, IMO.

**Pull request merged**